lib/ztp_helper.py

```diff
@@ -183,7 +188,31 @@ def download_file(self, file_url, destination_folder):
                 with open(destination_path, "w") as local_file:
                     for chunk in self.read_in_chunks(f):
                         local_file.write(chunk)
-                
+                        # Update md5sum everytime the file is being written
+                        hash_md5.update(chunk)
+
+                md5sum_local = hash_md5.hexdigest()
+
+                if md5sum:
+                    self.syslogger.info("MD5 Sum of the downloaded file is: %s" % hash_md5.hexdigest())
+
+                    if self.debug:
```

Comment:
Since this debug information, you might also want to print the user specified md5sum parameter (The remote md5sum) before you compare them,so that it becomes easier to debug on the router.

Reply:
@mvshpavan